test/test_entry_filter.rb

```diff
@@ -19,7 +19,7 @@ class TestEntryFilter < JekyllUnitTest
       files = %w[index.html site.css .htaccess vendor]
 
       @site.exclude = excludes + ["exclude*"]
-      assert_equal files, @site.filter_entries(excludes + files + ["excludeA"])
+      assert_equal files, @site.reader.filter_entries(excludes + files + ["excludeA"])
```

Comment:
This would require decoupling of the Reader and EntryFilter from the site. I do think this should be done, but should be left for future work, as it is quite some work and this pull request will become too large in my opinion.